docs/integrations/pyth.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Pyth
+head:
+  - - meta
+    - name: title
+      content: Solana Cookbook | Using Pyth to get OnChain data
+  - - meta
+    - name: og:title
+      content: Solana Cookbook | Using Pyth to get OnChain data
+  - - meta
+    - name: description
+      content: Pyth is an Oracle used to get real-word financial data onChain.
+  - - meta
+    - name: og:description
+      content: Pyth is an Oracle used to get real-word financial data onChain.
+  - - meta
+    - name: og:image
+      content: https://solanacookbook.com/cookbook-sharing-card.png
+  - - meta
+    - name: og:image:alt
+      content: Solana splash card
+  - - meta
+    - name: twitter:card
+      content: summary
+  - - meta
+    - name: twitter:site
+      content: "@solanacookbook"
+  - - meta
+    - name: twitter:image
+      content: "https://solanacookbook.com/cookbook-sharing-card.png"
+  - - meta
+    - name: robots
+      content: index,follow,noodp
+  - - meta
+    - name: googlebot
+      content: index,follow
+---
+
+# Pyth
+
+Pyth is an Oracle used to get real-world financial and crypto market data. Pyth Oracle can be used by on-chain programs in consuming data for a variety of use cases.
+
+## How to use Pyth in Client
+
+Pyth provides a JavaScript/TypeScript library called **@pythnetwork/client**. This library can be used to read on-chain Pyth Data for off-chain applications, such as displaying the Pyth price on a website. Learn more about this [here](https://www.npmjs.com/package/@pythnetwork/client)
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/pyth/client/client.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/pyth/client/client.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+## How to use Pyth in Anchor
+
+Pyth provides a Rust Crate which can be used by on-chain programs or off-chain application's to consume pyth's data.
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="Anchor" active>
+
+  <template v-slot:default>
+
+@[code](@/code/pyth/on-chain/on-chain.en.rs)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/pyth/on-chain/on-chain.preview.en.rs)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+## Other Resources
+
+- [Client Libraries](https://docs.pyth.network/consumers/client-libraries)
+- [JS Example Code](https://github.dev/solana-labs/solana/tree/master/web3.js/examples)
+- [Rust Example Code](https://github.com/project-serum/anchor/tree/master/tests/pyth)
+- [Anchor Example Code](https://github.com/0xPratik/pyth-anchor-example)
```

Comment:
Could you just make the anchor code on the cookbook example work vs linking to another repo? Otherwise people copying and pasting won't be able to run it

Reply:
Should l remove the **Anchor Example Code** as i thought of adding it because there are test cases which can help in running the repo quickly and exploring Pyth or should i add more code in the Code Block for Anchor like Test cases So that people can copy paste it and it works ?

Reply:
All full source examples should be copy-pastable and work as is. Recommend putting what you have in `lib.rs` on your example repo in instead of what is currently there. Just make sure you clean up the comments and any extraneous code

Reply:
Got it @jacobcreech working on it

Reply:
Hey @jacobcreech I have updated the Code and now the code in the repo and the code snippet in the cookbook matches and can work properly. I have tested it out.